src/index.js

```diff
@@ -1,49 +1,118 @@
 const {
   BaseKonnector,
   requestFactory,
+  log,
+  scrape,
   saveFiles,
-  addData
+  errors
 } = require('cozy-konnector-libs')
-const request = requestFactory({ cheerio: true })
+const request = requestFactory({
+  // debug: true,
+  cheerio: true,
+  jar: true,
+  json: false
+})
+const querystring = require('querystring')
 
-const baseUrl = 'http://books.toscrape.com'
+const baseUrl = 'https://cfspart.impots.gouv.fr'
 
 module.exports = new BaseKonnector(start)
 
-// The start function is run by the BaseKonnector instance only when it got all the account
-// information (fields). When you run this connector yourself in "standalone" mode or "dev" mode,
-// the account information come from ./konnector-dev-config.json file
-function start(fields) {
-  // The BaseKonnector instance expects a Promise as return of the function
-  return request(`${baseUrl}/index.html`).then($ => {
-    // cheerio (https://cheerio.js.org/) uses the same api as jQuery (http://jquery.com/)
-    // here I do an Array.from to convert the cheerio fake array to a real js array.
-    const entries = Array.from($('article')).map(article =>
-      parseArticle($, article)
-    )
-    return addData(entries, 'io.cozy.books').then(() =>
-      saveFiles(entries, fields)
-    )
-  })
+async function start (fields) {
+  await login(fields)
+  const documents = await fetch()
+  await saveFiles(documents, fields)
 }
 
-// The goal of this function is to parse a html page wrapped by a cheerio instance // and return an array of js objects which will be saved to the cozy by addData (https://github.com/cozy/cozy-konnector-libs/blob/master/docs/api.md#module_addData)
-// and saveFiles (https://github.com/cozy/cozy-konnector-libs/blob/master/docs/api.md#savefiles)
-function parseArticle($, article) {
-  const $article = $(article)
-  const title = $article.find('h3 a').attr('title')
-  return {
-    title,
-    price: normalizePrice($article.find('.price_color').text()),
-    url: `${baseUrl}/${$article.find('h3 a').attr('href')}`,
-    // when it finds a fileurl attribute, saveFiles will save this file to the cozy with a filename
-    // name
-    fileurl: `${baseUrl}/${$article.find('img').attr('src')}`,
-    filename: `${title}.jpg`
+async function login (fields) {
+  log('info', 'Logging in')
+  await request(`${baseUrl}/LoginMDP`)
```

Comment:
Why is this request necessary ? They have a cookie-based CSRF ? 

Anyway, may be it should be in the try catch so as to return `VENDOR_DOWN` if the site is down.

Reply:
This actually is not necessary. It was a precaution because some websites need some data in the cookie before submitting the authentication request but this is not the case here. I remove it.